### DISCUSSION TITLE
General Discussions

### DISCUSSION THREAD

**Discussion by michaldrabik** [trakt/showly#168] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded.

**Comment by michaldrabik** [comment]
Testing out new Github feature and opening first discussion :)

**Comment by cococoot** [comment]
![Screenshot_20201222-143817_Showly 20.jpg](https://user-images.githubusercontent.com/61015333/102926669-6f054700-4463-11eb-8e15-17599dbf54a4.jpg)

The checkmark does not need the white box around it. It will look better without the box. Just my thoughts. Great app.

**Reply by michaldrabik** [reply]
Thanks :) Maybe, but I wanted this to follow other outline buttons design (like the ones in single show details screens).

**Reply by DurchEins** [reply]
I think the checkboxes are pretty nice. Sometimes something is different than usual.

**Reply by michaldrabik** [reply]
Have not heard many bad words about them too but nobody has yet created a perfect UI that everyone would like 100%. It's understandable :)

**Comment by cococoot** [comment]
I understand. Would you consider making the box(s) throughout the app
smaller?  Thanks.

**Reply by michaldrabik** [reply]
Which boxes do you mean? Buttons?

**Comment by cococoot** [comment]
Look at the attached pics. I circled what I mean. It would look better if
you removed the white rectangle boxes and just left the white check mark.
If you don't want to do that make the white rectangle into a smaller
square. I tried to show in one pic but I'm no artist.

Also, can you add an option in settings to have the date and time a show
starts like this...

12/21/2020 8:00 PM

Same with movie premiers...

12/25/2020

Also, can you add in settings to alert us of a new episode 5 minutes before
show starts.

Thank you very much for the great app.

David
Show quoted text
[image: Error Icon]
Message too large
Your message couldn't be delivered to
*reply@reply.github.com* because it
exceeds the size limit. Try reducing the message size and resending.
The response from the remote server was:

552 5.3.4 Message size exceeds fixed limit



---------- Forwarded message ----------
From: David Scherrep <reserve1971@gmail.com>
To: "michaldrabik/showly-2.0" <
reply@reply.github.com>
Cc: David Scherrep <reserve1971@gmail.com>
Bcc:
Date: Wed, 23 Dec 2020 14:08:42 -0500
Subject: Re: [michaldrabik/showly-2.0] General Discussions (#168)
----- Message truncated -----

**Reply by michaldrabik** [reply]
Hmm I think images did not load.

**Comment by cococoot** [comment]
Look at the attached pics. I circled what I mean. It would look better if you removed the white rectangle boxes and just left the white check mark. If you don't want to do that make the white rectangle into a smaller square. I tried to show in one pic but I'm no artist. 


Also, can you add an option in settings to have the date and time a show starts like this...


12/21/2020 8:00 PM


Same with movie premiers...


12/25/2020


Also, can you add in settings to alert us of a new episode 5 minutes before show starts. 


Thank you very much for the great app. 


David

![IMG_20201223_140235.jpg](https://user-images.githubusercontent.com/61015333/103096380-bd385880-45d1-11eb-9bfc-9f688a065e33.jpg)

![IMG_20201223_140101.jpg](https://user-images.githubusercontent.com/61015333/103096388-c6292a00-45d1-11eb-9d4b-115347ccb935.jpg)

![IMG_20201223_140020.jpg](https://user-images.githubusercontent.com/61015333/103096397-cf19fb80-45d1-11eb-9231-6efc058ac5bb.jpg)

![IMG_20201223_135955.jpg](https://user-images.githubusercontent.com/61015333/103096403-d4774600-45d1-11eb-91f6-18c4e4a72dd3.jpg)

**Comment by cococoot** [comment]
I went back to GitHub and posted. Looks like the pics showed up there. Let
me know if you got them. Thank you.

**Reply by michaldrabik** [reply]
Yea. I don't think I personally like making these buttons smaller. These are main action buttons for those screens so I want them to be distinct and easliy clickable even on phones with high resolutions.

The dates format could be added I suppose. Also adding another time for notifications should be fine as well.

**Comment by cococoot** [comment]
Awesome, thanks!

**Comment by cococoot** [comment]
Just checking if you will make the updates I requested. Thank you.

David

**Reply by michaldrabik** [reply]
Hey. Sorry, I'm alone in this project and have many requests + my own plans for features + a daily job so I need to handle all this :) Thanks for understanding. Additional dates format will be added at some point just don't know when as I'm working on other stuff ATM.

**Comment by cococoot** [comment]
I understand. Thank you.

**Comment by kooolarpan** [comment]
Would love to see comment section where we can comment for a episode, movies which would sync to trakt.

**Reply by michaldrabik** [reply]
I do have these requests under #35 and #36 so commenting will appear eventually.

**Comment by seldon1000** [comment]
I'm not very familiar with Trak.tv, so I'm asking: would it be possible to display watched history for episodes and movies?

**Reply by michaldrabik** [reply]
In the app itself it's not yet possible but you can see your full history here: https://trakt.tv/users/me/history

**Reply by seldon1000** [reply]
Great, thank you! Are you planning to add history in the app?

**Reply by michaldrabik** [reply]
I'm not sure! I'll have to consider if this would be used by majority of users. I don't want to bloat the app with tons of advanced features which ultimately would kill its biggest strength - the simplicity.

**Reply by seldon1000** [reply]
Actually I rarely feel the need to check it, but sometimes it can be useful. I think the simplest way to add it into the app would be to add a separate tab in Progress section, next to Calendar. The layout could be nearly identical to Calendar's. However, it's not a must-have feature, the decision is up to you.

**Comment by kooolarpan** [comment]
I am watching movies/Series using Stremio which has Trakt integration too. In Showly, I have to manually add movies that I have watched. It is possible to automatically add to movies and series to Showly?

**Reply by michaldrabik** [reply]
I don't think it would be possible to sync it live with the app itself. Stremio <-> Trakt is probably a backend communication.
You simply need to sync though and the changes will appear!

**Reply by seldon1000** [reply]
Would scheduled sync do the trick? Setting it in order to import from Trak.tv once a day should be enough. It's not live though.

**Comment by seldon1000** [comment]
In 'Collection' screen there's a section for the recently added shows/movies. Sometimes, titles are hard to read with light thumbnails. Screenshot below.


![Screenshot_20210207-213919_Showly_2 0](https://user-images.githubusercontent.com/55358113/107158855-03ccf100-698d-11eb-975e-5179ffb0216e.png)


It's nothing serious, but I think it would be better to add a dark gradient at the bottom of the thumbnail to make the white text more readable. Or, maybe, something similar to the screenshot below: add a 'box' below the thumbnail which will contain the title, coloring it accordingly to the predominant color of the thumbnail.

![Screenshot_20210207-213136_Abbey](https://user-images.githubusercontent.com/55358113/107158793-b18bd000-698c-11eb-9e1c-aafd28f704fd.png)

I think it looks pretty cool this way.

**Reply by michaldrabik** [reply]
Hey - I considered both of these that you mentioned and finally decided to go with black shadow behind letters. I liked it best and as you said those cases with white backgrounds are pretty rare.

**Reply by seldon1000** [reply]
That's great! I like both solutions, actually the shadow does not take any more space and would be better for small screens. Thank you.

**Comment by seldon1000** [comment]
Hey, I've found myself using the "archive" feature quite a lot and I think it's extremely useful. However, I noticed that the archive only works within Showly, but no changes are applied to my Trakt account. This means that there's a lot of inconsistency (I have tens of shows in the archive) between Showly and my Trakt dashboard and, if for whatever reason I have to reinstall Showly, I would have to find and put all those shows back into the archive.

I also noticed that Trakt does provide the option to hide a show from my dashboard (progress, calendar...), and there's a separate section in my account settings called "Hidden Items" which contains all the show I've hidden. Now, before opening a potentially useless issue, I know nothing about Trakt APIs, so I'm asking if it would be possible to sync Showly's "Archive" with Trakt's "Hidden Items".

**Reply by michaldrabik** [reply]
Hey, yes the Archive was an idea just for Showly because there is not really this kind of section in Trakt. It's not synced currently.

There is a hidden items section in Trakt but I felt like it was not meant for everyday usage (it's pretty hidden).
I'm not sure if mixing Archive with hidden items would not be to confusing. Perhaps a better solution would be to sync it with a special custom list in Trakt that could have been called "Showly Archive" or something.

I may think about it after I'm done with custom lists.

**Reply by seldon1000** [reply]
Yeah, that section in Trakt is a bit hidden, but I mentioned it because it just seems to the the trick (I actually don't like how it's implemented in Trakt, TV Time did a much better job in this and Showly does the same but without sync). However, syncing the archive with a custom list could work, as long as the shows in that list stay hidden from progress and calendar screens (both on Showly and Trakt), otherwise it will be even more confusing and totally pointless.

Unfortunately, Trakt can be quite a mess, and this an example. One can only work on what the platform has to offer.

I see you're working hard on custom lists, so let's just talk about this for the moment.

**Comment by kooolarpan** [comment]
How does Showly shows the time of the release an episode? Does the time changes depending upon the time zone that is in Trakt? ![IMG_20210625_191715.jpg](https://user-images.githubusercontent.com/33162706/123432369-0b163380-d5ea-11eb-8060-48494b5cc7be.jpg)

**Reply by michaldrabik** [reply]
Date is taken from Trakt API where it's saved in UTC standard ISO format. For example this LOKI episode S01E04 has an air date of `2021-06-30T07:00:00.000Z` which is then converted to your local timezone when displayed.

**Reply by kooolarpan** [reply]
> Date is taken from Trakt API where it's saved in UTC standard ISO format. For example this LOKI episode S01E04 has an air date of `2021-06-30T07:00:00.000Z` which is then converted to your local timezone when displayed.

Oh, is the local time zone is fetched from information that is in Trakt ?

**Reply by michaldrabik** [reply]
No, It's what you have currently set on your phone.

**Reply by kooolarpan** [reply]
Thanks for the information.

**Comment by iam0day** [comment]
Congratulations, great app! Good job!

**Comment by conradseba** [comment]
Hey. I see the app getting better over time, so congrats on that.
I still think the white titles on top of thumbnails is a problem in many cases. I'm not saying you cannot read it, but is not nice either.
But what I wanted to point out here, is that the actors roles are missing. This is normally not a big issue since you can recognise them in live action titles, but it's a problem specially for animated titles. Who plays who in Sing2, or Kunfu Panda, etc?
Cheers.
![photo1641558127](https://user-images.githubusercontent.com/10448765/148544389-45afa59a-8d8b-4a80-8291-8cdcb4d8ff24.jpeg)

**Reply by michaldrabik** [reply]
Hi! Thanks for your input. At this moment you should be able to see character for the actor after you tap his/her portrait.

**Reply by conradseba** [reply]
ooohhh, shoot, right. Sorry for that! wonderful.

**Comment by ghost** [comment]
Hi, idk if this is the right place to ask this, but I see that this project have a nice project architecture and a good coding level so, my question is, where do you start learning or what were you doing until you reached this level of programming? I want to create an app but my code and architecture are always horrendous.

**Reply by esQmo** [reply]
@Relwi Practice I guess? :D The more you code, the more your level improves itself

**Reply by michaldrabik** [reply]
Thank you. There are still so many refractors waiting to be done to make the codebase decent in my eyes :)

Answering question - I think it's just years of coding and practice. I've been doing this as my career for quite some years by now. I guess the best advice is just keep coding and make new stuff. Each project will be better than previous one.

**Reply by ghost** [reply]
Thanks guys! Time to practice and make some projects then :)

**Comment by ghost** [comment]
Hi, is me again, I saw your project and other projects and you distributres the packages in other projects, like in this app example, you put the "ui-*" packages as if they where other projects, and my question is, why you do that? Is there any advantages doing like this or is only to distribute better the code?

**Reply by michaldrabik** [reply]
Hi! This is just a convention to distinguish modules more easily in a multi-module app. The UI prefix indicates that this module contains UI related part of the app. Something similar can be found here https://github.com/chrisbanes/tivi

**Comment by alebupal** [comment]
Hello, first of all congratulations for the app, it's wonderful!

Would it be possible to be able to select the text of the comments?
My main language is not English and there are times when I wish I could translate a comment. As the text is not selectable I cannot copy and paste in the translator.

I hope you understand, thank you very much and greetings!

**Comment by ghost** [comment]
[Suggestion] Material You design on Showly

**Reply by iam0day** [reply]
It would be nice, but in my opinion the current UI is easier to maintain.... Although with Flutter it would definitely have the default Material You theme without too much effort.

**Comment by Fooftilly** [comment]
Would you be open to adding a way to buy premium outside of the play store, because it is not available in some countries? Something like Netguard is doing. I would appreciate it because I would like to have premium features and to support the app.

**Reply by michaldrabik** [reply]
@Fooftilly It's not possible via PlayStore officialy due to Google policies.  Feel free to drop me an email showlyapp@gmail.com and we can figure something out. Where are you from?

**Reply by Fooftilly** [reply]
I've sent you an email.
